mode/libraries/ar/build.gradle

```diff
@@ -2,47 +2,60 @@ import java.nio.file.Files
 import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
 
 apply plugin: 'aar'
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
 
 dependencies {
     compileOnly name: "android"
-
     compileOnly "org.p5android:processing-core:${modeVersion}"
-
     implementationAar "com.google.ar:core:${garVersion}"
 }
 
-task createPom {
-    pom {
-       project {
-           groupId "org.p5android"
-           artifactId "processing-ar"
-           version "${arLibVersion}"
-           packaging "jar"
-           licenses {
-               license {
-                   name "GNU Lesser General Public License, version 2.1"
-                   url "https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt"
-                   distribution "repo"
-               }
-           }
-           dependencies {
-               dependency {
-                   groupId "org.p5android"
-                   artifactId "processing-core"
-                   version "${modeVersion}"
-                   scope "implementation"
-               }
+task sourceJar(type: Jar) {
+    from sourceSets.main.allJava
+    archiveClassifier = "sources"
+}
+
+publishing {
+    publications {
+        arPublication(MavenPublication) {
+            from components.java
+            artifact sourceJar
+            pom {
+                groupId = "org.p5android"
+                artifactId = "processing-ar"
+                version = "${arLibVersion}"
+                packaging = "jar"
+                // description = "Processing Android Core"
+                // url = "http://www.example.com/project"
+                licenses {
+                    license {
+                        name = "GNU Lesser General Public License, version 2.1"
+                        url = "https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt"
+                        distribution = "repo"
+                    }
+                }
+            }
+
+            pom.withXml {
+                // inserting the dependencies node
+                def dependenciesNode = asNode().appendNode('dependencies')
+                // start adding dependency nodes inside dependencies node
+                def dependencyNode = dependenciesNode.appendNode('dependency')
+                dependencyNode.appendNode('groupId', 'org.p5android')
+                dependencyNode.appendNode('artifactId', 'processing-core')
+                dependencyNode.appendNode('version', "${modeVersion}")
+                dependencyNode.appendNode('scope', 'implementation')
+
+                def dependencyNode2 = dependenciesNode.appendNode('dependency')
+                dependencyNode2.appendNode('groupId', 'com.google.ar')
```

Comment:
Use proper names for the variables !

PS:  as mentioned above about the naming of variables.

Reply:
Corrected variable names in AR build.gradle